[MISC] Add minimal Dockerfile

yuxiang-gao opened the pull request:
This pull request adds a minimal workable Dockerfile to run Genesis.

To run it, first build the docker:

```bash
docker build -t genesis .
```

Then  run it using the following command:

```bash
xhost +

docker run --gpus all --rm -it -e DISPLAY=$DISPLAY -v /tmp/.X11-unix/:/tmp/.X11-unix genesis python examples/tutorials/hello_genesis.py
```

zhouxian commented:
Thank you! Could you add something to readme with explanations?

yuxiang-gao commented:
> Thank you! Could you add something to readme with explanations?

Sure, happy to. I'll ping you once I'm finished.

antoinebou12 commented:
Can you add a docker-compose

```
services:
  genesis:
    build:
      context: .
      dockerfile: Dockerfile
    image: genesis:latest
    runtime: nvidia
    environment:
      - DISPLAY=${DISPLAY}
    volumes:
      - /tmp/.X11-unix:/tmp/.X11-unix
      - ./data:/app/data  # Example volume; adjust as needed
    command: python examples/tutorials/hello_genesis.py
    deploy:
      resources:
        reservations:
          devices:
            - capabilities: [gpu]
```

yuxiang-gao commented:
> Can you add a docker-compose
> 
> https://github.com/ETSTribology/Genesis/blob/main/Dockerfile
> 
> my version
> 
> ```
> services:
>   genesis:
>     build:
>       context: .
>       dockerfile: Dockerfile
>     image: genesis:latest
>     runtime: nvidia
>     environment:
>       - DISPLAY=${DISPLAY}
>     volumes:
>       - /tmp/.X11-unix:/tmp/.X11-unix
>     command: python examples/tutorials/hello_genesis.py
>     deploy:
>       resources:
>         reservations:
>           devices:
>             - capabilities: [gpu]
> ```

Thanks! I have included this version of compose file to this pull request.

yuxiang-gao commented:
Hi @zhouxian , can you take a look? I have updated both the English and Chinese version of READMEs. Also added a `docker-compose.yml` file and instructions thanks to @antoinebou12 .

antoinebou12 commented:
You could also make a action to auto deploy to dockerhub

zhouxian commented:
@Kashu7100 any idea how we proceed with this?

YilingQiao commented:
Thanks for your contribution! However, we already have another commit for the Dockerfile. We will acknowledge your help in our next release notes!
